Library/Homebrew/formula_auditor.rb

```diff
@@ -438,13 +438,16 @@ def audit_bottle_spec
 
     def audit_bottle_disabled
       return unless formula.bottle_disabled?
-      return if formula.bottle_unneeded?
 
-      problem "Unrecognized bottle modifier" unless formula.bottle_disable_reason.valid?
-
-      return unless @core_tap
-
-      problem "Formulae in homebrew/core should not use `bottle :disabled`"
+      if !formula.bottle_disable_reason.valid?
+        problem "Unrecognized bottle modifier"
+      elsif @core_tap
+        if formula.bottle_unneeded?
+          problem "Formulae in homebrew/core should not use `bottle :unneeded`"
+        else
+          problem "Formulae in homebrew/core should not use `bottle :disabled`"
+        end
+      end
```

Comment:
An optional suggestion:

```suggestion
      problem "Unrecognized bottle modifier" unless formula.bottle_disable_reason.valid?
      
      return unless @core_tap

      problem "Formulae in homebrew/core should not use `bottle :unneeded`" if formula.bottle_unneeded?
      problem "Formulae in homebrew/core should not use `bottle :disabled`" if formula.bottle_disabled?
```

Reply:
@bayandin don't think this will quite work as `. bottle_disabled?` is `true` when `.bottle_unneeded?` is true

Reply:
Ah, right, sorry for the noise then. I had to have a look at the implementation first 😳

Reply:
@bayandin No apologies needed! I'd rather these suggestions even if they won't work over not getting them at all! ❤️